docs/topics/exceptions.md

```diff
@@ -1,122 +1,679 @@
+    }
+    if (amount < 1 || amount % 1 != 0.0) {
+        throw WithdrawalException("Invalid withdrawal amount.")
+    }
+    println("Withdrawal processed")
+}
+
+fun main() {
+    val availableFunds = 500.0
+
+    // Change this value to test different scenarios
+    val withdrawalAmount = 500.5
+
+    try {
+        processWithdrawal(withdrawalAmount.toDouble(), availableFunds)
+
+    // The order of catch blocks is important!
+    } catch (e: InsufficientFundsException) {
+        println("Caught an InsufficientFundsException: ${e.message}")
+    } catch (e: WithdrawalException) {
+        println("Caught a WithdrawalException: ${e.message}")
+    }
+}
 ```
+{kotlin-runnable="true"}
 
-The returned value of a `try` expression is either the last expression in the `try` block or the
-last expression in the `catch` block (or blocks).
-The contents of the `finally` block don't affect the result of the expression.
+When you have multiple `catch` blocks, it's important to order them from the most specific to the least specific exception,
+following a top-to-bottom order in your code. 
+This ordering aligns with the program's execution flow. 
+If you were to catch a general exception before a more specific one, the specific exception would never be reached, 
+as the general catch block would intercept all exceptions, including the specific ones.
 
-## Checked exceptions
+### The finally block
 
-Kotlin does not have checked exceptions. There are many reasons for this, but we will provide a simple example that illustrates why it is the case.
+The `finally` block is a block of code that is always executed, regardless of whether the `try` block succeeds without
+any problems or an exception is thrown.
+The primary use of the `finally` block is to clean up code after the execution of `try` and `catch` block(s).
+Using `finally` ensures that resources acquired in the `try` block are always released, preventing potential resource leaks.
+It plays a critical role when working with resources like files or network connections, ensuring that they are properly
+closed or released.
 
-The following is an example interface from the JDK implemented by the `StringBuilder` class:
+Here is how you would typically use the `try-catch-finally` blocks together:
 
-``` java
-Appendable append(CharSequence csq) throws IOException;
+```kotlin
+try {
+    // code that may throw an exception
+}
+catch (e: YourException) {
+    // exception handler
+}
+finally {
+    // code that is always executed
+}
 ```
 
-This signature says that every time I append a string to something (a `StringBuilder`, some kind of a log, a console, etc.),
-I have to catch the `IOExceptions`. Why? Because the implementation might be performing IO operations (`Writer` also implements `Appendable`).
-The result is code like this all over the place:
+The returned value of a `try` expression is determined by the last expression in the `try-catch` block that gets executed.
```

Comment:
Please simplify the paragraph.

Reply:
how about this one:

The returned value of a `try` expression is determined by the last executed expression in either the `try` or `catch` block. If no exceptions occur, the result comes from the `try` block; if an exception is handled, it comes from the `catch` block. The `finally` block is always executed, but it doesn't change the result of the `try-catch` block.

Reply:
Keep your version